### DISCUSSION TITLE
is it possible to reset the vdev counter number when recreating the vdev again

### DISCUSSION THREAD

**Discussion by comsyspro** [openzfs/zfs#12063] [Q&A]
when you initially create a pool with let's say 2 vdevs in raidz1 then the 2 vdevs are named raidz1-0 and raidz1-1. now when you destroy raidz1-1 and recreate it again the newly created vdev will be named  raidz1-**2**. now when you destroy raidz1-2 and recreate it again the newly created vdev will be named  raidz1-**3**... so my question is if it is possible to reset the vdev counter so if you recreate the vdevs the counter will not grow or you will be able to reset the number by a command.

**Comment by rincebrain** [comment]
You can't remove raidz1 vdevs, and even if this were about mirror vdevs, there is not presently a way to reset that counter, because IIRC vdev evacuation "secretly" still has the vdev listed with a special do-not-allocate flag, so the number of that vdev is still used.

**Comment by comsyspro** [comment]
thanks for clarification.
